cli/src/commands/log.rs

```diff
@@ -127,19 +134,36 @@ pub(crate) fn cmd_log(
                 // the missing edges since we don't care about where they point here
                 // anyway?
                 let mut has_missing = false;
+                let mut elided = vec![];
                 for edge in edges {
                     match edge.edge_type {
                         RevsetGraphEdgeType::Missing => {
                             has_missing = true;
                         }
-                        RevsetGraphEdgeType::Direct => graphlog_edges.push(Edge::Present {
-                            direct: true,
-                            target: edge.target,
-                        }),
-                        RevsetGraphEdgeType::Indirect => graphlog_edges.push(Edge::Present {
-                            direct: false,
-                            target: edge.target,
-                        }),
+                        RevsetGraphEdgeType::Direct => {
+                            graphlog_edges.push(Edge::Present {
+                                direct: true,
+                                target: edge.target,
+                            });
+                        }
+                        RevsetGraphEdgeType::Indirect => {
+                            if use_elided_nodes {
+                                let mut synthetic_commit_id_bytes = edge.target.to_bytes();
+                                // Let's hope this commit id doesn't exist
+                                synthetic_commit_id_bytes.push(0);
+                                let synthetic_commit_id = CommitId::new(synthetic_commit_id_bytes);
+                                elided.push((synthetic_commit_id.clone(), edge.target));
+                                graphlog_edges.push(Edge::Present {
+                                    direct: true,
+                                    target: synthetic_commit_id,
```

Comment:
Perhaps, synthetic target could be `Real(CommitId) | Synthetic(CommitId)` enum, but that would require extra cloning. I'm fine with either way.

Reply:
Oh, that reminds me that I meant to try switching to a `(CommitId, bool)` pair as target instead. I've done that. Let me know if you prefer an enum.